### PR TITLE
Requests endpoint

### PR DESCRIPTION
**Pull request by JordanNguyen**
finished requests endpoint + tests to follow our new structure (model validation, strong parameters)

### CODE REVIEW

File: dispatchr-ruby/app/controllers/requests_controller.rb [hunk 1]
```diff
@@ -12,56 +8,46 @@ def index
...
-			exp_date = params[:expiration_date]
-			request = Request.new(user_id: user_id, expiration_date: exp_date)
-
-			if request.save
-				render json: request
-			else
-				render json: request.errors, status: :unprocessable_entity
-			end
-
+		request = Request.new(
```
Comment (alokg95) [inline dispatchr-ruby/app/controllers/requests_controller.rb]: Why not just do Request.new(request_params)?

File: dispatchr-ruby/spec/controllers/requests_controller_spec.rb [hunk 2]
```diff
@@ -24,30 +26,80 @@
...
 				expect(response).to have_http_status(200)
 			end
+
 			it 'has the correct user_id and expiration_date' do
-				post :create, {user_id: 1, expiration_date: '0'}
+				create_request
 				request = Request.last
 				expect(request.user_id).to eq(1)
-				expect(request.expiration_date).to eq(nil)
+				expect(request.expiration_date).to eq("0028-11-16T00:00:00.000Z")
```
Comment (alokg95) [inline dispatchr-ruby/spec/controllers/requests_controller_spec.rb]: This spec will start failing after the 28th. Why not do a farther away expiration date?

Comment (JordanNguyen) [inline dispatchr-ruby/spec/controllers/requests_controller_spec.rb]: nah it should be fine. i've put dates that were in the past and it still worked

File: dispatchr-ruby/app/controllers/requests_controller.rb (hunk 1 above)
Comment (JordanNguyen) [inline dispatchr-ruby/app/controllers/requests_controller.rb]: I was just following your controller. you didnt use user_params ?

File: dispatchr-ruby/spec/controllers/requests_controller_spec.rb (hunk 2 above)
Comment (alokg95) [inline dispatchr-ruby/spec/controllers/requests_controller_spec.rb]: Oh okay, cool

File: dispatchr-ruby/app/controllers/requests_controller.rb (hunk 1 above)
Comment (alokg95) [inline dispatchr-ruby/app/controllers/requests_controller.rb]: What do you mean? I haven't merged addresses branch btw.

Comment (JordanNguyen) [inline dispatchr-ruby/app/controllers/requests_controller.rb]: nvm i was thinking of something else lol

**Comment by JordanNguyen** [comment]
[Requests API setup (4)](https://trello.com/c/5LCyhPfk/15-requests-api-setup-4)

**Pull request merged**